src/modules/core/components/SingleUserPicker/utils.ts

```diff
@@ -1,34 +1,32 @@
+import { AnyUser } from '~data/index';
+
 /*
 Extracts the required values to be used in the SingleUserPicker
 on selection
 */
-export const filterUserSelection = (data, filterValue) => {
+export const filterUserSelection = (data: AnyUser[], filterValue: string) => {
   if (!filterValue) {
     return data;
   }
 
-  const filtered = data.filter(
-    (user) =>
-      user &&
-      user.profile &&
-      user.profile.username &&
-      filterValue &&
-      (user.profile.username
-        .toLowerCase()
-        .includes(filterValue.toLowerCase()) ||
-        user.profile.walletAddress
-          .toLowerCase()
-          .includes(filterValue.toLowerCase()) ||
-        user.id.toLowerCase().includes(filterValue.toLowerCase())),
-  );
+  const filteredUsers = data.filter((user) => {
+    const username = user.profile?.username || '';
+    const walletAddress = user.profile?.walletAddress || '';
```

Comment:
I mean... the wallet address must always exist, but in this case we're also not excluding the entry if there isn't one.

I think we should filter out entries which don't have a wallet address _(but you are correct in allowing entries without a username)_

Granted, what I just explained above shouldn't really happen in real life

Reply:
Yeah. now that you mentioned it, it _shouldn't_ happen that a user doesn't have a wallet address. 

With that being said, I'll add an early return in the case of the object not having a wallet address, just for safeguarding.